openadapt/record.py

```diff
@@ -618,12 +631,29 @@ def record(
 
     # TODO: discard events until everything is ready
 
+    collect_stats()
+
     try:
         while True:
             time.sleep(1)
     except KeyboardInterrupt:
         terminate_event.set()
 
+    collect_stats()
+
+    stats = performance_snapshots[-1].compare_to(performance_snapshots[-2], 'lineno')
```

Comment:
Can you please clarify why we're only comparing the second-to-last to the last? Is it because the number of expected snapshots is two? In that case, what do you think about something like this:

```
assert len(performance_snapshots) == 2, performance_snapshots
first_snapshot, last_snapshot = performance_snapshots
...
```

Reply:
The expected number is two. I've added your code in [e96b942](https://github.com/MLDSAI/OpenAdapt/pull/270/commits/e96b9420bd9ad3bdf15e2f7561cfea451eb06797)